### PR TITLE
feat(Resource Access Management): Edit Resource Access Rule implementation

### PR DESCRIPTION
**Pull request by dmyios**
## Description

[screen-capture (3).webm](https://github.com/centreon/centreon/assets/108675430/9be86db9-2bdc-4269-84db-3efa2db1439d)

1. Implemented Edit Resource Access Rule part (form, API calls, tests)
2. Fixed update script 23.10.5
3. Updated form to prototype v3

**Fixes** # MON-34502

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/actions/lighthouse-performance-testing/action.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ runs:
         php-version: 8.1
         extensions: yaml, xml, mysql, dom, mbstring, intl, pdo, zip
         coverage: none
-        tools: composer:v2
+        tools: composer:v2.6
```
Comment (jeremyjaouen) [inline .github/actions/lighthouse-performance-testing/action.yml:44]:
```suggestion
        tools: composer:2.6
```

File: centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInitialValues/initialValues.ts [hunk 2]
```diff
@@ -1,25 +1,62 @@
...
   contacts: [],
-  datasetFilters: [[{ resourceType: undefined, resources: [] }]],
+  datasetFilters: [[{ resourceType: ResourceTypeEnum.Empty, resources: [] }]],
   description: '',
   isActivated: true,
   name: ''
 });
 
+const formatDatasetFilter = (datasetFilter: DatasetFilter): Array<Dataset> => {
+  let datasets: Array<Dataset> = [];
```
Comment (Thebarda) [inline centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInitialValues/initialValues.ts:21]: Use constant instead
```suggestion
  const datasets: Array<Dataset> = [];
```

File: centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInitialValues/initialValues.ts [hunk 3]
```diff
@@ -1,25 +1,62 @@
...
+  while (!equals(prop('datasetFilter', datasetFilter), null)) {
+    datasets = [
+      ...datasets,
+      {
+        resourceType: datasetFilter.resourceType,
+        resources: datasetFilter.resources
+      }
+    ];
+    datasetFilter = prop('datasetFilter', datasetFilter);
+  }
```
Comment (Thebarda) [inline centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInitialValues/initialValues.ts:31]: use map instead

**Review by sc979 — APPROVED** [review]
pipeline fix LGTM

File: centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInitialValues/initialValues.ts (hunk 2 above)
Comment (dmyios) [reply centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInitialValues/initialValues.ts]: done

File: centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInitialValues/initialValues.ts (hunk 3 above)
Comment (dmyios) [reply centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInitialValues/initialValues.ts]: done

**Review by Thebarda — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**